contracts/UniswapV2Router01.sol

```diff
@@ -3,21 +3,23 @@ pragma solidity =0.6.6;
 import '@uniswap/v2-core/contracts/interfaces/IUniswapV2Factory.sol';
 import '@uniswap/lib/contracts/libraries/TransferHelper.sol';
 
+import './libraries/UniswapV2Library.sol';
 import './interfaces/IUniswapV2Router01.sol';
-import './UniswapV2Library.sol';
 import './interfaces/IERC20.sol';
 import './interfaces/IWETH.sol';
 
-contract UniswapV2Router01 is IUniswapV2Router01, UniswapV2Library {
+contract UniswapV2Router01 is IUniswapV2Router01 {
     address public immutable override WETH;
+    address public immutable override factory;
```

Comment:
nit: let's make factory the first address here, in the constructor definition + body, and in the interface